kingfisher_scrapy/spiders/paraguay_dncp_base.py

```diff
@@ -45,17 +45,33 @@ def from_crawler(cls, crawler, *args, **kwargs):
         return spider
 
     def start_requests(self):
-        url = f'{self.base_url}/search/processes?tipo_fecha=fecha_release&' \
-              f'fecha_desde={self.from_date.strftime(self.date_format)}-04:00&' \
-              f'fecha_hasta={self.until_date.strftime(self.date_format)}-04:00'
-
-        yield self.build_request(
-            url,
-            formatter=parameters('fecha_desde'),
-            # send duplicate requests when the token expired and in the continuation of last_request saved.
-            dont_filter=True,
-            callback=self.parse_pages
-        )
+        for url in self.urls_builder():
+            yield self.build_request(
+                url,
+                formatter=parameters('fecha_desde'),
+                # send duplicate requests when the token expired and in the continuation of last_requests saved.
+                dont_filter=True,
+                callback=self.parse_pages
+            )
+
+    def urls_builder(self):
+        days_interval = 30
+        # ElasticSearch doesn't allow searches sizes bigger than 10000.
+        # To avoid that, we request one month at the time.
+        days_to_iterate = timedelta(days=days_interval)
+        end_date = self.until_date
+        start_date = self.from_date
+        # In reverse chronological order
+        while end_date > self.from_date:
+            if (end_date - start_date).days > days_interval:
+                start_date = (end_date - days_to_iterate).replace(hour=0, minute=0, second=0)
+            else:
+                start_date = self.from_date
+            url = f'{self.base_url}/search/processes?tipo_fecha=fecha_release&' \
+                  f'fecha_desde={start_date.strftime(self.date_format)}-04:00&' \
+                  f'fecha_hasta={end_date.strftime(self.date_format)}-04:00&items_per_page=500'
+            end_date = (start_date - timedelta(1)).replace(hour=23, minute=59, second=59)
```

Comment:
Use the keyword argument for clarity.

```suggestion
            end_date = (start_date - timedelta(days=1)).replace(hour=23, minute=59, second=59)
```